src/Engines/nativeEngine.ts

```diff
@@ -1765,7 +2005,13 @@ export class NativeEngine extends Engine {
             return false;
         }
 
-        this._native.setFloat4(uniform, x, y, z, w);
+        this._commandBufferEncoder.beginEncodingCommand(this._native.COMMAND_SETFLOAT4);
+        this._commandBufferEncoder.encodeCommandArgAsUInt32(uniform);
+        this._commandBufferEncoder.encodeCommandArgAsFloat32(x);
+        this._commandBufferEncoder.encodeCommandArgAsFloat32(y);
+        this._commandBufferEncoder.encodeCommandArgAsFloat32(z);
+        this._commandBufferEncoder.encodeCommandArgAsFloat32(w);
+        this._commandBufferEncoder.finishEncodingCommand();

```

Comment:
(9This refers to all of the calls to `this._commandBufferEncoder`) can all of these calls be done by a single function accepting optional parameters? seems like those calls are constantly being repeated but with different parameters

Reply:
If I understand, the idea would be to be able to pass multiple args to `encodeCommandArgAsFloat32` (for example), and the function would have to check if those values were actually passed in before trying to encode them into the buffer, is that right? If so, I think we could try it, we'd just have to see if it impacts perf at all.